sklearn/model_selection/_search.py

```diff
@@ -866,6 +867,33 @@ def _get_scorers(self):
 
         return scorers, refit_metric
 
+    def _check_scorers_accept_sample_weight(self):
+        # TODO(slep006): remove when metadata routing is the only way
+        scorers, _ = self._get_scorers()
+        # In the multimetric case, warn the user for each scorer separately
+        if isinstance(scorers, _MultimetricScorer):
+            for name, scorer in scorers._scorers.items():
+                if not scorer._accept_sample_weight():
+                    warnings.warn(
+                        f"The scoring {name}={scorer} does not support sample_weight, "
+                        "which may lead to statistically incorrect results when "
+                        f"fitting {self} with sample_weight. "
+                    )
```

Comment:
Note: I think we should find a way to issue a similar warning when metadata routing is enabled. But I think we can keep the scope of this PR to the case where it is disabled and implement a solution in a subsequent, once the default routing policy is implemented.

Note that we have a similar problem with the warning raised by `CalibratedClassifierCV`.